packages/atomic/src/components/atomic-results-per-page-v1/atomic-results-per-page.tsx

```diff
@@ -0,0 +1,129 @@
+  @InitializeBindings() public bindings!: Bindings;
+  private resultPerPage!: ResultsPerPage;
+  public searchStatus!: SearchStatus;
+  private choices!: number[];
+
+  @State()
+  @BindStateToController('resultPerPage')
+  public resultPerPageState!: ResultsPerPageState;
+  @BindStateToController('searchStatus')
+  @State()
+  private searchStatusState!: SearchStatusState;
+  @State() public error!: Error;
+
+  /**
+   * A list of choices for the number of results to display per page, separated by commas.
+   */
+  @Prop() choicesDisplayed = '10,25,50,100';
+  /**
+   * The initial selection for the number of result per page. This should be part of the `choicesDisplayed` option. By default, this is set to the first value in `choicesDisplayed`.
+   */
+  @Prop({mutable: true}) initialChoice?: number;
+
+  public initialize() {
+    this.choices = this.validateChoicesDisplayed();
+    this.validateInitialChoice();
+
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
+    this.resultPerPage = buildResultsPerPage(this.bindings.engine, {
+      initialState: {numberOfResults: this.initialChoice},
+    });
+  }
+
+  private validateChoicesDisplayed() {
+    return this.choicesDisplayed.split(',').map((choice) => {
+      const parsedChoice = parseInt(choice);
+      if (isNaN(parsedChoice)) {
+        const errorMsg = `The choice value "${choice}" from the "choicesDisplayed" option is not a number.`;
+        this.bindings.engine.logger.error(errorMsg, this);
+        throw new Error(errorMsg);
+      }
+
+      return parsedChoice;
+    });
+  }
+
+  private validateInitialChoice() {
+    if (!this.initialChoice) {
+      this.initialChoice = this.choices[0];
+      return;
+    }
+    if (!this.choices.includes(this.initialChoice)) {
+      const errorMsg = `The "initialChoice" option value "${this.initialChoice}" is not included in the "choicesDisplayed" option "${this.choicesDisplayed}".`;
+      this.bindings.engine.logger.error(errorMsg, this);
+      throw new Error(errorMsg);
+    }
+  }
+
+  private buildChoice(choice: number) {
+    const isSelected = this.resultPerPage.isSetTo(choice);
+
+    return (
+      <Button
+        style="outline-neutral"
+        role="radio"
+        ariaChecked={`${isSelected}`}
+        ariaLabel={this.bindings.i18n.t('display-results-per-page', {
+          results: choice,
+        })}
+        onClick={() => this.resultPerPage.set(choice)}
+        class={`number-button ${isSelected ? 'selected' : ''}`}
+        part={`button${isSelected && ' active-button'}`}
+        text={choice.toLocaleString(this.bindings.i18n.language)}
+      ></Button>
+    );
+  }
+
+  public render() {
+    if (!this.searchStatusState.hasResults) {
+      return;
+    }
+
+    return (
+      <div class="flex items-center">
+        <span part="label" class="text-on-background text-lg mr-3">
+          {this.bindings.i18n.t('results-per-page')}
+        </span>
+        <div
```

Comment:
The [standard keyboard shortcuts}(https://www.w3.org/TR/2016/WD-wai-aria-practices-1.1-20160317/examples/radio/radio.html#id_kbd) for a radiogroup should probably be supported here.

Reply:
I don't like it cause every switch triggers a new query which takes time and changes the page height, it's not like a form with a few radio button where going between each of them does not affect the interface in an important way.

Nevertheless, you are right and we should support it! Also the tab should only get you to the selected elements not the others